qemu/tests/netkvm_protocol_binding.py

```diff
@@ -58,9 +66,19 @@ def run(test, params, env):
 
     test.log.info("Will install inf file found at '%s'", inf_path)
     install_cmd = params["install_cmd"] % inf_path
-    status, output = session.cmd_status_output(install_cmd, timeout=timeout)
-    if status:
-        test.error("Install inf file failed, output=%s" % output)
+
+    # workaround for driver no signature
+    key_to_install_driver = params.get("key_to_install_driver").split(";")
+    thread_installation = utils_misc.InterruptedThread(session.cmd_status_output, (install_cmd, timeout))
+    thread_installation.start()
+    time.sleep(10)
+    thread_sendkey = utils_misc.InterruptedThread(send_key, (vm, key_to_install_driver))
```

Comment:
This could be normally sent from main thread instead of opening another one, isn't it?

Reply:
Yes, it makes sense. My initial thought was to use two threads to make the code look more logical.And the join the installation thread to main thread and wait for it finished.